src/main/java/com/myproject/service/impl/UserServiceImpl.java

```diff
@@ -22,9 +23,13 @@
     @Autowired
     private BasketDao basketDao;
 
+    @Autowired
+    private BCryptPasswordEncoder passwordEncoder;
+
     @Transactional
     @Override
-    public void add(User user) {
+    public void add(User user, String nonHashedPass) {
+        user.setHashedPassword(passwordEncoder.encode(nonHashedPass));
```

Comment:
can we write?
user.setHashedPassword(passwordEncoder.encode(user.getPassword));

Reply:
@Sofasmile I think no. Because if we talk about user, he has only hashedPass, and if in registration controller we set in hashedPass nonHashedPass it is not good. But, maybe I am wrong.